lms/static/scripts/frontend_apps/components/test/FilePickerApp-test.js

```diff
@@ -104,6 +114,81 @@ describe('FilePickerApp', () => {
     });
   }
 
+  context('when create assignment configuration is enabled', () => {
+    const authURL = 'https://testlms.hypothes.is/authorize';
+    const createAssignmentPath = '/api/canvas/assignments';
+    let fakeAPICall;
+    let fakeNewAssignment;
+
+    beforeEach(() => {
+      fakeConfig.filePicker.createAssignmentAPI = {
+        authURL,
+        path: createAssignmentPath,
+      };
+
+      fakeAPICall = sinon.stub();
+      fakeNewAssignment = { ext_lti_assignment_id: 10 };
+
+      fakeAPICall
+        .withArgs(sinon.match({ path: createAssignmentPath }))
+        .resolves(fakeNewAssignment);
+
+      $imports.$mock({
+        '../utils/api': { apiCall: fakeAPICall },
+      });
+    });
+
+    it('calls backend api when content is selected', async () => {
+      const onSubmit = sinon.stub().callsFake(e => e.preventDefault());
+      const wrapper = renderFilePicker({ onSubmit });
+
+      selectContent(wrapper, 'https://example.com');
+
+      await waitFor(() => fakeAPICall.called);
+      await delay(100);
```

Comment:
This is the last change since you last looked at it @robertknight 

Not sure if necessary of a symptom of something else that's missing/wrong.